hubbard/hamiltonian.py

```diff
@@ -692,6 +707,10 @@ def spin_contamination(self, ret_exact=False):
         S: float
             exact expectation value
         """
+
+        if self.size<2:
+            raise ValueError('To obtain the spin contamination the Hamiltonian needs to be spin-polarized!')
```

Comment:
Is this necessary? If the user insists on doing this, the answer is just zero spin contamination, no?

Reply:
Well I thought that it didn't make much sense to obtain the spin contamination for the unpolarized case, but I can modify this function to work also for this case and return the zero spin contamination.

Reply:
I think that would be better than raising errors. I see it more as a trivial calculation (rather than an error).

Reply:
Done! See commit 75d58b9